feedbacks/views.py

```diff
@@ -1,43 +1,93 @@
-from django.views.generic.list import ListView
+from django.views.generic.detail import DetailView
 from django.views.generic.edit import DeleteView, UpdateView, CreateView
 from django.contrib.auth.mixins import LoginRequiredMixin
-from django.core.paginator import EmptyPage, Paginator
 from django.urls import reverse_lazy
-from feedbacks.models import Feedback
 
-ITEMS_ON_PAGE = 5
+from CultureAnalyzer.settings.base_settings import ITEMS_ON_PAGE
+from CultureAnalyzer.view import SafePaginationListView
+from feedbacks.models import Feedback, Recommendation
+from feedbacks.forms import FeedbackForm, RecommendationForm
 
+__all__ = [
+    'FeedbackListView',
+    'FeedbackCreateView',
+    'FeedbackUpdateView',
+    'FeedbackDeleteView',
+    'FeedbackDetailView',
+    'RecommendationCreateView',
+    'RecommendationUpdateView',
+    'RecommendationDeleteView',
+]
 
-class SafePaginator(Paginator):
-    def validate_number(self, number):
-        try:
-            return super(SafePaginator, self).validate_number(number)
-        except EmptyPage:
-            if number > 1:
-                return self.num_pages
-            else:
-                return 1
 
-
-class FeedbackListView(LoginRequiredMixin, ListView):
+class FeedbackListView(LoginRequiredMixin, SafePaginationListView):
     model = Feedback
-    paginator_class = SafePaginator
     paginate_by = ITEMS_ON_PAGE
     context_object_name = 'feedbacks'
     ordering = ['id']
 
 
+class FeedbackDetailView(LoginRequiredMixin, DetailView):
+    model = Feedback
+
+
 class FeedbackDeleteView(LoginRequiredMixin, DeleteView):
     model = Feedback
     success_url = reverse_lazy('feedback-list')
 
 
 class FeedbackCreateView(LoginRequiredMixin, CreateView):
     model = Feedback
-    fields = ['feedback']
+    form_class = FeedbackForm
 
 
 class FeedbackUpdateView(LoginRequiredMixin, UpdateView):
     model = Feedback
-    fields = ['feedback']
+    form_class = FeedbackForm
+    template_name_suffix = '_form'
+
+
+class RecommendationDeleteView(LoginRequiredMixin, DeleteView):
+    model = Recommendation
+
+    def delete(self, request, *args, **kwargs):
+        """Redirect to linked feedback"""
+        back = self.get_object().feedback.id
+        self.success_url = reverse_lazy('feedback-detail', kwargs={'pk': back})
+        print(back)
+        return super().delete(self, request, *args, **kwargs)
```

Comment:
You have 2 requests to DB: `get` & `delete`. Those requests should be run inside transaction only!